concepts/identity.md

```diff
@@ -0,0 +1,283 @@
+  identifiers relating to the participant (eg. META-ID = [Spotify-ID, IPI,
+  National ID, PGP key, Twitter handle])
+
+## Terminology
+
+*TODO*: complete definitions of terms used in this document
+
+- Custodian
+- Smart Contract
+- Identity Owner
+- META-ID
+- Identity Signature Scheme
+- Identity Custody Scheme
+- Trust Item - a detached cryptographic signature of a metadata item
+
+sha3() indicates keccak256 hashing of the contents within parentheses.
+
+## Core concepts
+
+A Meta Identity MUST define:
+
+- META-ID: A 32 byte ENS namehash
+- CID: A content-addressed identifier
+
+The META-ID is the UNIQUE identifier of the Meta Identity. It is IMMUTABLE. There are no constraints on the hash data. The META-ID is ONLY an index key, and MUST NOT be used for cryptographic purposes.
+
+The CID MUST point to a valid META Object. The META Object MUST include a property "meta." The value of this property MUST be a data structure as defined in subsection "Core" under "Metadata."
+
+The manifest CAN include an arbitrary number of metadata items. These define claims of auxiliary identifiers linked to the Meta Identity. These SHOULD define data structures as defined in subsection "Auxiliary" under "Metadata."
+
+### Ownership and permissions
+
+A Meta Identity MUST be owned by one single Ethereum address. The mapping of address to Meta Identity MUST be recorded on the blockchain. Ownership of the Meta Identity MUST be enforced through the Identity Custody Scheme. The initial Meta Identity owner SHOULD be the transaction sender of the Meta Identity instantiation.
+
+This ownership CAN be transferred to a different Ethereum address. Activation of this feature SHOULD be optional, and MUST include a user-modifiable grace period during which reverting to previous ownership is possible.
+
+A Meta Identity owner CAN perform transfer of ownership. A Meta Identity owner CAN appoint a number of Ethereum addresses to act as *custodians*. These CAN through a vote perform transfer of ownership. A vote MUST require a message signed by a respective private key. The *minimum number of addresses* that must make up a custodian role, and *minimum ratio of votes* required to execute a transfer CAN ONLY be determined by the Meta Identity owner.
+
+### Implementation
+
+
+The Meta Identity uses ENS as backend for storage and retrieval, interfaced by the Identity and IdentityResolver contracts.
+
+The Identity contract is upgradeable and MUST be registered in the IdentityController contract. As the IdentityController instantiates all permanent storage, and guarantees that the contents of these cannot be altered between upgrades. Therefore only the instance of the Identity contract that has been registered with the IdentityController can be considered valid for use.
+
+In the following we assume the META-ID to be `sha3("foo")`.
+
+#### Add a new identity
+
+```
+Identity.register(bytes32 metaidhash, address owner);
```

Comment:
Variables should be [mixedCase](http://solidity.readthedocs.io/en/develop/style-guide.html#local-and-state-variables) (so `metaIdHash` rather than `metaidhash`)

Reply:
@nolash I see you've addressed this on a branch of meta-network/identity but it looks like only some of the variables have been fixed. We can update them in the doc here but just a note to reflect this in the contracts.